### PR TITLE
Add cancel SetupIntent

### LINKED ISSUE #266: Cancel SetupIntent
Hi!

Is it possible to [cancel](https://stripe.com/docs/api/setup_intents/cancel) a SetupIntent? I see cancel in [PaymentIntent](https://github.com/arlyon/async-stripe/blob/master/src/resources/payment_intent_ext.rs#L34), but not for SetupIntent.

### PR DESCRIPTION
**Pull request by ViktorWb**
Hi!
This implements cancel for SetupIntents. The API is very similar to cancel for PaymentIntents.

Fixes #266.

### CODE REVIEW

**Comment by tl-christopher-fleetwood** [comment]
LGTM

**Comment by FL33TW00D** [comment]
LGTM

**Comment by arlyon** [comment]
Good stuff, running CI now, will merge if it's green :)

**Comment by arlyon** [comment]
We recently added some checks to ensure the codegen is unchanged in PRs however it seems to be overactive. I will investigate. Thanks again for this! :)

File: src/resources/setup_intent_ext.rs [hunk 1]
```diff
@@ -32,6 +39,17 @@ impl SetupIntent {
     ) -> Response<SetupIntent> {
         client.post_form(&format!("/setup_intents/{}/confirm", setup_id), &params)
     }
+
```
Comment (mzeitlin11) [inline src/resources/setup_intent_ext.rs:42]:
```suggestion

```
I think `fmt` is removing this extra line causing the CI issue. Running `cargo make fmt` should fix it

**Comment by arlyon** [comment]
Ah yes that seems to be the case. Let's split the ci in two and run formatting first so that we fail early. I can write a PR @mzeitlin11 or I can defer to you, let me know what works best!

**Pull request merged**